lib/chrome.js

```diff
@@ -183,7 +190,14 @@ export class StorageArea {
   }
 
   clear(callback) {
-    this.items.clear().then(function() {if (callback) {callback();}});
+    this.items.list()
+    .then(records => Promise.all(records.data.map(record =>
+      this.items.delete(record.id))))
```

Comment:
We want the operation to result in all records being deleted on the server as well.